torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -132,6 +190,12 @@ class TORCH_API RpcAgent {
   // by terminating all RPC threads.
   virtual void shutdown() = 0;
 
+  // Function that cleans up the local state for RpcAgent. This is called from
+  // the destructor, and ensures that we can gracefully destruct even if the
+  // child class was not successfully constructed without calling a virtual
+  // function.
+  void cleanup();
```

Comment:
maybe make it private, if there's no need/desire for external entities to call this?